### PR TITLE
Added the oauth2 rest template for epis calls with the client_credent…

### PR DESCRIPTION
**Pull request by ErkoRisthein**
…ials token

### CODE REVIEW

**Comment by ErkoRisthein** [comment]
Example usage:
```
    private final OAuth2RestOperations oAuth2RestTemplate;

    @Value("${epis.service.url}")
    private String episServiceUrl;

    ...

    ResponseEntity<FundDto[]> response = oAuth2RestTemplate.getForEntity(episServiceUrl + "/funds", FundDto[].class);
```

File: src/main/java/ee/tuleva/onboarding/config/Oauth2RestTemplateConfiguration.java [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+public class Oauth2RestTemplateConfiguration {
+
+    @Bean
+    @ConfigurationProperties("epis.service.client")
+    ClientCredentialsResourceDetails oauth2ClientCredentialsResourceDetails() {
+        return new ClientCredentialsResourceDetails();
+    }
+
+    @Bean
+    public OAuth2RestOperations oauth2RestTemplate(ClientCredentialsResourceDetails resource, OAuth2ClientContext context) {
```
Comment (v3rm0n) [inline src/main/java/ee/tuleva/onboarding/config/Oauth2RestTemplateConfiguration.java]: I'm wondering if `EnableOAuth2Client` and `OAuth2ClientContext` are actually needed here 🤔 Doesn't it work without them?

Comment (ErkoRisthein) [reply src/main/java/ee/tuleva/onboarding/config/Oauth2RestTemplateConfiguration.java]: Not sure, but it's taken from the official spring javadocs: https://docs.spring.io/spring-security/oauth/apidocs/org/springframework/security/oauth2/config/annotation/web/configuration/EnableOAuth2Client.html

Comment (v3rm0n) [reply src/main/java/ee/tuleva/onboarding/config/Oauth2RestTemplateConfiguration.java]: This is for Authorization Code Grant, which definitely needs oauth context.

Comment (ErkoRisthein) [reply src/main/java/ee/tuleva/onboarding/config/Oauth2RestTemplateConfiguration.java]: You're right, it does work without the `@EnableOAuth2Client` and `OAuth2ClientContext` here. 🙇

**Pull request merged**